tests/scheduled_task/scheduled_task_suite_test.go

```diff
@@ -0,0 +1,237 @@
+
+			step = "creating a datacenter resource with 2 nodes"
+			testFile, err := ginkgo_util.CreateTestFile(dcYaml)
+			Expect(err).ToNot(HaveOccurred())
+
+			k = kubectl.ApplyFiles(testFile)
+			ns.ExecAndLog(step, k)
+
+			ns.WaitForDatacenterReady(dcName)
+
+			step = "creating a ScheduledTask for cleanup"
+
+			k = kubectl.ApplyFiles(scheduledTaskYaml)
+			stdout, stderr, err := ns.ExecVCapture(k)
+			Expect(err).ToNot(HaveOccurred())
+			fmt.Println(stdout)
+			fmt.Println(stderr)
+
+			step = "verify ScheduledTask is created"
+			json = "jsonpath={.metadata.name}"
+			k = kubectl.Get(scheduledTaskResource).FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, scheduledTaskName, 60)
+
+			step = "verify ScheduledTask status fields are populated"
+			json = "jsonpath={.status.nextSchedule}"
+			k = kubectl.Get(scheduledTaskResource).FormatOutput(json)
+			// Wait for nextSchedule to be set (non-empty)
+			Eventually(func() string {
+				output, err := ns.Output(k)
+				if err != nil {
+					return ""
+				}
+				return output
+			}, time.Minute*2, time.Second*5).ShouldNot(BeEmpty())
+
+			step = "verify ScheduledTask creates CassandraTasks"
+			json = "jsonpath={.items[*].metadata.name}"
+			k = kubectl.Get("CassandraTask").
+				WithLabel(dcLabel).
+				FormatOutput(json)
+
+			// Wait for at least one CassandraTask to be created
+			Eventually(func() string {
+				output, err := ns.Output(k)
+				if err != nil {
+					return ""
+				}
+				return output
+			}, time.Minute*5, time.Second*10).ShouldNot(BeEmpty())
+
+			step = "verify CassandraTask has correct command"
+			json = "jsonpath={.items[0].spec.jobs[0].command}"
+			k = kubectl.Get("CassandraTask").
+				WithLabel(dcLabel).
+				FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, "cleanup", 60)
+
+			step = "verify CassandraTask references correct datacenter"
+			json = "jsonpath={.items[0].spec.datacenter.name}"
+			k = kubectl.Get("CassandraTask").
+				WithLabel(dcLabel).
+				FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, dcName, 60)
+
+			step = "wait for first CassandraTask to complete"
+			json = "jsonpath={.items[0].status.completionTime}"
+			k = kubectl.Get("CassandraTask").
+				WithLabel(dcLabel).
+				FormatOutput(json)
+
+			Eventually(func() string {
+				output, err := ns.Output(k)
+				if err != nil {
+					return ""
+				}
+				return output
+			}, time.Minute*10, time.Second*10).ShouldNot(BeEmpty())
+
+			step = "verify LastExecution is updated after task completion"
+			json = "jsonpath={.status.lastExecution}"
+			k = kubectl.Get(scheduledTaskResource).FormatOutput(json)
+
+			Eventually(func() string {
+				output, err := ns.Output(k)
+				if err != nil {
+					return ""
+				}
+				return output
+			}, time.Minute*2, time.Second*5).ShouldNot(BeEmpty())
+
+			step = "verify multiple CassandraTasks are created over time with ForbidConcurrent policy"
+			// Wait a bit more for the next scheduled execution
+			time.Sleep(70 * time.Second)
```

Comment:
No sleeps, this is why we have Eventually in those WaitFor functions

Reply:
I can move the Get into the eventually here indeed to remove the sleep.